src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -144,4 +162,38 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
             _OpenJsonHandlers(nullptr, target);
         }
     }
+
+    void MainPage::_InitializeProfilesList()
+    {
+        // Manually create a NavigationViewItem for each profile
+        // and keep a reference to them in a map so that we
+        // can easily modify the correct one when the associated
+        // profile changes.
+        for (auto profile : _settingsSource.AllProfiles())
+        {
+            MUX::Controls::NavigationViewItem profileNavItem;
+            profileNavItem.Content(box_value(profile.Name()));
+            profileNavItem.Tag(box_value<guid>(profile.Guid()));
+
+            const auto iconSource{ IconPathConverter::IconSourceWUX(profile.Icon()) };
+            WUX::Controls::IconSourceElement icon;
+            icon.IconSource(iconSource);
+            profileNavItem.Icon(icon);
+
+            SettingsNav().MenuItems().Append(profileNavItem);
+            _profileToNavItemMap.Insert(profile.Guid(), profileNavItem);
+        }
+
+        // Top off (the end of the nav view) with the Add Profile item
+        MUX::Controls::NavigationViewItem addProfileItem;
+        addProfileItem.Content(box_value(RS_(L"Nav_AddNewProfile/Content")));
+        addProfileItem.Tag(box_value(L"AddNewProfile_Nav"));
+
+        FontIcon icon;
+        // This is the "Add" symbol
+        icon.Glyph(L"\xE710");
+        addProfileItem.Icon(icon);

```

Comment:
gotta make sure the add profile one isn't actually selectable, and does something else.. right?